ktor-server/ktor-server-cio/jvmAndNix/src/io/ktor/server/cio/backend/HttpServer.kt

```diff
@@ -22,30 +22,52 @@ import kotlinx.coroutines.*
 public fun CoroutineScope.httpServer(
     settings: HttpServerSettings,
     handler: HttpRequestHandler
+): HttpServer {
+    val selector = SelectorManager(coroutineContext)
+    return httpServer(
+        createServer = {
+            aSocket(selector).tcp().bind(
+                hostname = settings.host,
+                port = settings.port
+            )
+        },
+        serverJobName = CoroutineName("server-root-${settings.port}"),
+        acceptJobName = CoroutineName("accept-${settings.port}"),
+        timeout = WeakTimeoutQueue(settings.connectionIdleTimeoutSeconds * 1000L),
+        selector = selector,
+        handler = handler
+    )
+}
+
+/**
+ * Start an http server with [settings] invoking [handler] for every request
+ */
+@InternalAPI
+public fun CoroutineScope.httpServer(
+    createServer: () -> ServerSocket,
+    timeout: WeakTimeoutQueue,
+    serverJobName: CoroutineName,
+    acceptJobName: CoroutineName,
+    selector: SelectorManager,
```

Comment:
This argument can be eliminated. Right now it's only for:
```kotlin
    serverJob.invokeOnCompletion {
        selector.close()
    }
```
but `serverJob` is available in the returned `HttpServer` instance as `rootServerJob`.

So the caller can do
```kotlin
val selector = ...
val server = httpServer(...)
server.rootServerJob.invokeOnCancelation {
  selector.cancel()
}
```